src/attribute.rs

```diff
@@ -324,6 +327,25 @@ fn bytes_to_vec(val: Vec<u8>) -> Vec<u8> {
 }
 conversion_from_type! {make from_bytes; from_type_bytes; from_string_bytes; from Vec<u8>; as BytesType; via bytes_to_vec}
 
+pub fn string_from_sized(t: CK_ULONG, val: &[u8]) -> Attribute {
+    let mut value = Vec::from(val);
+    let mut len = value.len();
+    for i in (0..len).rev() {
+        if value[i] != 0x20 {
+            break;
+        }
```

Comment:
why not simply convert to a string and then use [trim_end()](https://doc.rust-lang.org/std/string/struct.String.html#method.trim_end) ?

Reply:
I wanted to be explicit about what we trim. The `trim_end()` removes all the whitespaces, newlines and other stuff, but PKCS#11 is specific that only the 0x20 byte should be used for padding.

I could theoretically use `trim_end_matches()` for that as, but I thought it would require a conversion to string, which I wanted to avoid.

Reply:
ok, good enough for me